### DISCUSSION TITLE
Debugging

### DISCUSSION THREAD

**Discussion by rvwilliams88** [unoplatform/uno#17245] [General]
When I run my simple app (still learning) the screen displays _An error occurred_.  How can I find out what was the error that caused this message? Is there a general error handler I can switch off?

**Comment by jeromelaban** [comment]
Thanks for the report. Can you provide a screenshot of the error, as well as the environment and IDEs that you're using?

**Comment by rvwilliams88** [comment]
I'm on Windows 11 and using Visual Studio 2022 17.10.2.  The message is exactly as I described - An error has occurred - nothing more and nothing less.

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by jeromelaban** [reply]
Thank you. Can you provide a screenshot of the error? We're not able to help if we do not know where the error comes from.

**Reply by jeromelaban** [reply]
> Here's the screenshot.

Sending emails to conversations does not attach email, unfortunately. Can you attach it though the web UI?

**Comment by rvwilliams88** [comment]
I suspect that the compiler is including a catch-all exception handler. This can be shown by putting a 'throw' in otherwise working code - the same message then occurs. What I need to know is how to switch that off.

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by rvwilliams88** [comment]
Here's the screenshot.
***@***.***

Regards

Richard.
****************************************************
Richard Williams
email: ***@***.******@***.***>
mobile: 07981 734697
****************************************************

**Comment by rvwilliams88** [comment]
screenshot.
![error screenshot](https://github.com/unoplatform/uno/assets/112261322/056e731b-26eb-4674-8f2b-92c628fab3de)

**Reply by jeromelaban** [reply]
_Please make sure to answer in the same thread, it makes following the discussion easier. Thanks!_

Thanks. This helps. This is likely a change from the code itself, if you can [provide a repro](https://platform.uno/docs/articles/uno-howto-create-a-repro.html) this will help, also, you can [break on all exceptions](https://stackoverflow.com/a/116934) and see what is raising this exception.

**Reply by jeromelaban** [reply]
Thanks for the repro, this is not a complete repro (a library is missing), so we're unable to determine the cause of this isue.

If you break on all exception, the origin of that exception will be useful, can you provide a screenshot of the stack trace?

**Reply by jeromelaban** [reply]
_Please make sure to answer in the same thread, it makes following the discussion easier. Thanks!_

> I've created a new branch called 'no-library' where all traces of my code library have been removed and replaced with a mocked-up list of Persons.  The error still occurs.  I've ticked every box in the Exceptions settings but this hasn't revealed any more information about the error.

When breaking on all exceptions, you'll notice this:

![image](https://github.com/unoplatform/uno/assets/5839577/cae89247-0d57-4f5a-b856-20ec6b66fc4c)

Which causes the `FeedView` to go in error state (the "An error has occurred" message).

SelectedPerson may be null, and this is something that would have been caught if the `!` [was not used here](https://github.com/rvwilliams88/PeopleApp/blob/65435382990f1253ad66ce0cd31214b1c601edfa/PeopleApp/PeopleModel.cs#L34). In general, it's best to keep the nullable types everywhere possible.

**Reply by jeromelaban** [reply]
> I still can't make VS 2022 break on all exceptions.  Which version of VS are you using, I'm on 17.10.3

This is a default VS feature, make sure to check this box:

![image](https://github.com/unoplatform/uno/assets/5839577/70e4266c-3acc-4ed3-a6e5-8b9bc3a0469f)


> Are there any samples available that show how to use the result of a Selection in the call to a further function (for example, in the PeopleApp sample to use the SelectedPerson as the parameter to another function rather than just used to generate a simple string)?

@dr1rrb may be able to answer.

**Reply by jeromelaban** [reply]
@rvwilliams88 Also could you avoid answering to this discussion using emails? Please use the github UI, or the mobile app. This make the discussion very difficult to follow.

**Comment by rvwilliams88** [comment]
Hi Jerome

Breaking on exceptions didn't help as the same message comes up.  I've made a repro on GitHub - rvwilliams88/PeopleApp.  This app is just a simple extension of the PeopleApp sample as described in https://platform.uno/docs/articles/external/uno.extensions/doc/Learn/Mvux/Advanced/Selection.html.  Changes I have made are

i)                    Model is now C# not XAML;

ii)                   The 'people' are loaded from an external database - the work being done by the Library workspace (the code for this cannot be revealed)

iii)                 ListView is now ComboBox.

What I'm trying to do is to take the SelectedPerson value and use that to extract further information from the database (mocked up with a return of list of strings).

The changes made in points i), ii) and iii) above all work as I've tested this separately.  The error is coming from my attempt to use the SelectedPerson value.

**Comment by rvwilliams88** [comment]
Hi Jérôme

I've created a new branch called 'no-library' where all traces of my code library have been removed and replaced with a mocked-up list of Persons.  The error still occurs.  I've ticked every box in the Exceptions settings but this hasn't revealed any more information about the error.

I must say that I probably wouldn't be having this problem if the section on Selection in the MVUX Docs had included an example of how to use the SelectedPerson value to look up more information rather than just showing how to created a simple text message.

Regards

Richard.
****************************************************
Richard Williams
email: ***@***.******@***.***>
mobile: 07981 734697
****************************************************

**Comment by rvwilliams88** [comment]
I still can't make VS 2022 break on all exceptions.  Which version of VS are you using, I'm on 17.10.3

Are there any samples available that show how to use the result of a Selection in the call to a further function (for example, in the PeopleApp sample to use the SelectedPerson as the parameter to another function rather than just used to generate a simple string)?

Regards

Richard.
****************************************************
Richard Williams
email: ***@***.******@***.***>
mobile: 07981 734697
****************************************************
